ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -317,7 +317,7 @@ public static BlobProperties buildBlobProperties(Map<String, Object> args) throw
       int sizeToAllocate = 0;
       for (Map.Entry<String, Object> entry : args.entrySet()) {
         String key = entry.getKey();
-        if (key.startsWith(Headers.USER_META_DATA_HEADER_PREFIX)) {
+        if (key.toLowerCase().startsWith(Headers.USER_META_DATA_HEADER_PREFIX)) {
           // key size
           sizeToAllocate += 4;
           String keyToStore = key.substring(Headers.USER_META_DATA_HEADER_PREFIX.length());
```

Comment:
So, the part of the key after the prefix will still be case sensitive once persisted in storage?

Reply:
I didn't want to change the case as received from the client. Also there is no "sensitivity" here since there are no equals/substring checks on the part of the key after the prefix.

Reply:
Yep, we don't want to change this behavior since clients might expect the key to be cased.